### PR TITLE
DEVPROD-3617: Allow users to specify a task to run on task generation.

### PR DESCRIPTION
**Pull request by johndaniels**
This adds an activate_generated_tasks parameter, which allows for specifying variant/task pairs as part of the version parameters. auto_task_all will generate all of those variant/task pairs with activate:true s that they are automatically run.

Thanks for submitting a PR to the Genny repo. Please include the following fields (if relevant) prior to submitting your PR.

**Jira Ticket:** DEVPROD-3617

**Whats Changed:**  
This adds an activate_generated_tasks parameter, which allows
for specifying variant/task pairs as part of the version
parameters. auto_task_all will generate all of those variant/task
pairs with activate:true so that they are automatically run.

**Patch testing results:**  
If applicable, link a patch test showing code changes running successfully

https://spruce.mongodb.com/version/659dcd8fa4cf477e674acaa1/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

array_traversal and large_scale_long_lived in the patch above were activated automatically by this code.

**Workload Submission form:**  
If applicable, only required if there is a new workload being added. Form can be found [here](https://docs.google.com/forms/d/1r0kOHvFUa7rJxVmX_wp9prxYU5K155yKyZ1y3d5yhZg/edit)

### CODE REVIEW

File: src/lamplib/src/genny/tasks/auto_tasks_all.py [hunk 1]
```diff
@@ -32,36 +36,52 @@ def get_all_builds(global_expansions: dict[str, str], project_file_path: str) ->
...
-        SLOG.error(f"Evergreen expansions file {os.path.join(workspace_root, 'expansions.yml')} does not exist. Ensure this file exists and that it is in the correct location.")
+        SLOG.error(
+            f"Evergreen expansions file {os.path.join(workspace_root, 'expansions.yml')} does not exist. Ensure this file exists and that it is in the correct location."
+        )
         sys.exit(1)
     execution = int(global_expansions["execution"])
+    activate_tasks = set(
+        VariantTask(*item.strip().split(":"))
+        for item in global_expansions.get("activate_generated_tasks", "").split(",")
+    )
```
Comment (SimonEismann) [inline src/lamplib/src/genny/tasks/auto_tasks_all.py:77]: While correct, this is a bit hard to parse. My personal preference would be to split this into 2-3 lines so it reads can be read top down, while right now it needs to be read "bottom-up". Feel free to ignore this.

File: src/lamplib/src/genny/tasks/auto_tasks_all.py [hunk 2]
```diff
@@ -11,13 +11,17 @@
...
 SLOG = structlog.get_logger(__name__)
 
-def get_all_builds(global_expansions: dict[str, str], project_file_path: str) -> List[CurrentBuildInfo]:
+
+def get_all_builds(
+    global_expansions: dict[str, str], project_file_path: str
+) -> List[CurrentBuildInfo]:
     """
-    Attempts to compute the expansions 
+    Attempts to compute the expansions
```
Comment (SimonEismann) [inline src/lamplib/src/genny/tasks/auto_tasks_all.py:24]: It seems your formatter disagrees with some of the existing formatting, which results in formatting-only changes like this that make the PR harder to parse.

My preference here would be that we either:
a) apply no formatter at all
b) set up an agreed-upon formatter as part of the CI tests (not as part of this ticket)

**Review by SimonEismann — APPROVED** [review]
Looks good, let's make sure to document this aggressively and announce it in #performance-tooling-users. Could this be used in some of our tools to schedule tasks (e.g., when kicking off profiling builds for generated tasks)? If so, let's open corresponding tickets.

File: src/lamplib/src/genny/tasks/auto_tasks_all.py (hunk 2 above)
Comment (johndaniels) [reply src/lamplib/src/genny/tasks/auto_tasks_all.py:24]: I attempted to what was set up as part of the repo, and explicitly ran `run-genny lint-python --fix`, which _should_ match what we do in CI. I'm a bit mystified by the fact that it fixed anything, but I figured I'd add it.

Comment (johndaniels) [reply src/lamplib/src/genny/tasks/auto_tasks_all.py:24]: That is, we're supposed to be doing 2 above but it seems like something is wonky.

File: src/lamplib/src/genny/tasks/auto_tasks_all.py (hunk 1 above)
Comment (johndaniels) [reply src/lamplib/src/genny/tasks/auto_tasks_all.py:77]: Agreed. Will clean up.

**Comment by johndaniels** [comment]
evergreen merge

**Comment by johndaniels** [comment]
evergreen merge
